oeps/oep-0031-arch-i18n.rst

```diff
@@ -0,0 +1,122 @@
+
+When choosing what locale to use for a page, the locale in the URL always takes precedence. If there is none, check the user's cookies for a locale; if there isn't one, fetch it from the user's account settings and store it in their cookies for reuse. If there is no locale (or no locale we support) in the account settings, we will use the first locale we can support in the ``Accept-Language`` header that was sent with the request. Failing everything else, we will fall back to English.
+
+Alternatives Considered
+-----------------------
+
+We chose to avoid putting the locale in the domain name itself, to leave that available for other uses. We considered but rejected putting it in as a URL parameter; that can be harder to parse out and doesn't give us automatic separation of analytics by locale. Using the first component of the path ensures that if users send a link or bookmark a page, their recipients will see the exact content that they intended to share.
+
+For the locale selection strategy, we considered but rejected a fallback step where we would choose a language based on the geographic location of the user's IP address. This is generally not reliable: many countries have speakers of multiple languages, users might be traveling, and users might be living or studying in a country but still be more comfortable in a different language.
+
+Accessibility Requirements
+==========================
+
+Translate all non-display text that is present in a tag for accessibility reasons (e.g., ``aria-label``), using the same locale as the rest of the page.
+
+There is a special case where a message id doesn't have a translation in the user's preferred locale, so we fall back to a default locale. In this case, that string needs to be wrapped in a ``<span locale="....">`` tag to maintain the correct locale information for accessibility tools. In order to make sure that this string wrapping always happens, edX has written ``FMessage``, a simple wrapper around the standard react-intl ``FormattedMessage`` component, which will be the standard component used to format strings. ``FMessage`` can be found in the **[TODO]** repo.
+
+Message Ids
+===========
+
+Message ids, or message keys, uniquely identify a string to be internationalized. They should consist of one or more words to scope the message, plus one or more words to identify the message. All words will be lowercase and separated by periods. Here are examples of message ids used in a header and in a login modal, respectively:
```

Comment:
Nit: Can we use the term "namespace" rather than "scope" here?  In usage, a scope has generally referred to a region of a program.  Here, you are not enforcing any scoping rules but recommending usage of namespaces.